app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -199,12 +199,23 @@ function normalizePPOMRequest(
   const { delegationMock, id, jsonrpc, method, origin, params } =
     normalizedRequest;
 
+  // In case of gator permissions the origin is always gator snap.
+  // That is why we extract the actual origin from the decoded permission.
+  let actualOrigin = origin;
+  if (
+    controllerObject &&
+    'decodedPermission' in controllerObject &&
+    controllerObject.decodedPermission?.origin
+  ) {
+    actualOrigin = controllerObject.decodedPermission.origin;
+  }
+
```

Comment:
This looks great - just a couple of thoughts:

Maybe we put this into `normalizeSignatureRequest` instead, which is where the `normalizedRequest` object originates. It feels preferable to not pollute this top level `normalizePPOMRequest` function with specific details of the normalization.

Although the `decodedPermission` should only ever be set if the requesting origin is the `gator-permissions-snap`, I wonder if it's worthwhile guarding this behaviour with `isSnapId` and `isSnapPreinstalled` checks on the actual request origin, because this is a critical piece of information. If an actor was able to somehow bypass checking the origin of the request, it would be catastrophic.